File: DailyChallenges/Scala/LeetCode_2279.scala
object Solution {
    def maximumBags(capacity: Array[Int], rocks: Array[Int], additionalRocks: Int): Int = {
        var ans = 0
        var extra = additionalRocks
        for (slot <- capacity.zip(rocks).map(t => t._1 - t._2).sorted if extra >= slot){
            extra -= slot
            ans += 1
        }
        ans
    }
}

File: DailyChallenges/CPP/LeetCode_2279.cpp
class Solution {
public:
    int maximumBags(vector<int>& capacity, vector<int>& rocks, int additionalRocks) {
        int n = capacity.size();
        priority_queue<int, vector<int>, greater<int>> pq;
        for (int i=0; i<n; ++i)
            pq.push(capacity[i]-rocks[i]);
        while(!pq.empty() && additionalRocks >= pq.top()){
            additionalRocks -= pq.top(); pq.pop();
        }
        return (n-pq.size());
    }
};
